session/pingpong/hermes_url_getter.go

```diff
@@ -84,19 +93,36 @@ func (hug *HermesURLGetter) GetHermesURL(chainID int64, address common.Address)
 		hug.loadedAddresses[chainID] = make(map[common.Address]string, 0)
 	}
 
-	registry, err := hug.addressProvider.GetRegistryAddress(chainID)
+	url, err := hug.getHermesURLBC(chainID, address)
+	if err != nil {
+		log.Err(err).Msg("failed to get hermes url from blockchain, using fallback")
+		url, err = hug.getHermesURLObserver(chainID, address)
+		if err != nil {
+			return "", err
+		}
+	}
+	url, err = hug.normalizeAddress(url)
 	if err != nil {
 		return "", err
 	}
+	hug.loadedAddresses[chainID][address] = url
+	return url, nil
+}
 
-	add, err := hug.bc.GetHermesURL(chainID, registry, address)
+func (hug *HermesURLGetter) getHermesURLBC(chainID int64, address common.Address) (string, error) {
+	registry, err := hug.addressProvider.GetRegistryAddress(chainID)
 	if err != nil {
 		return "", err
 	}
-	add, err = hug.normalizeAddress(add)
+
+	return hug.bc.GetHermesURL(chainID, registry, address)
+}
+
+func (hug *HermesURLGetter) getHermesURLObserver(chainID int64, address common.Address) (string, error) {
+	hermesData, err := hug.observer.GetHermesData(chainID, address)
 	if err != nil {
 		return "", err
 	}
-	hug.loadedAddresses[chainID][address] = add
-	return add, nil
+	log.Debug().Msgf("returned url from observer %s=%s", address.Hex(), hermesData.URL)
```

Comment:
Lets use `Fields()` or something instead of dynamic log message. Easier to grep if needed.

Reply:
it was leftover from debugging, I'll delete it!